dali/operators/image/resize/resize_attr.cc

```diff
@@ -22,66 +22,85 @@
-(same as ``size``). The coordinates can be either absolute (in pixels, the default) or
-relative (0..1), depending on the value of ``relative_roi`` argument. If a RoI origin is greater
-than RoI end in any dimension, the region is flipped in that dimension.))", nullptr, true)
-  .AddOptionalArg("roi_relative", R"(If true, RoI coordinates are relative to the input size, with
-0 denoting top/left and 1 denoting bottom/right)", false);
+the aspect ratio of the original volume. A negative value flips the volume.)code", 0.f, true)
+  .AddOptionalArg<vector<float>>("size", R"code(The desired output size. Must be a list/tuple
+with the one entry per spatial dimension (i.e. excluding video frames and channels).
+
+Dimensions with 0 extent are treated as absent and the output size will be calculated
+based on other extents and ``mode`` argument.)code", {}, true)
+  .AddOptionalArg("mode", R"code(Resize mode.
+
+Here is a list of the supported values:
+
+* | ``"default"`` - image is resized to the specified size.
+  | Missing extents are scaled with the average scale of the provided ones extents.
+* | ``"stretch"`` - image is resized to the specified size.
+  | Missing extents are not at all scaled.
+* | ``"not_larger"`` - image is resized, keeping the aspect ratio, so that no extent of the
+    output image exceeds the specified size.
+  | For example, a 1280x720, with a desired output size of 640x480, actually produces
+    a 640x360 output.
+* | ``"not_smaller"`` - image is resized, keeping the aspect ratio, so that no extent of the
+    output image is smaller than specified.
+  | For example a 640x480 imag,e with a desired output size of 1920x1080, actually produces
+    a 1920x1440 output.
+
+This argument is mutually exclusive with ``resize_longer`` and ``resize_shorter``)code", "default")
+  .AddOptionalArg("resize_shorter", R"code(The length of the shorter dimension of the resized image.
+
+This option is mutually exclusive with ``resize_longer`` and explicit size arguments, and
+the operator keeps the aspect ratio of the original image. This option is equivalent to specifying
+the same size for all dimensions and ``mode="not_smaller"``. The longer dimension can be bounded
+by setting the ``max_size`` argument. See ``max_size`` argument documentation for
+more information.)code", 0.f, true)
+  .AddOptionalArg("resize_longer", R"code(The length of the longer dimension of the resized image.
+
+This option is mutually exclusive with ``resize_shorter`` and explicit size arguments, and
+the operator keeps the aspect ratio of the original image. This option is equivalent to specifying
+the same size for all dimensions and ``mode="not_larger"``.
+)code", 0.f, true)
+  .AddOptionalArg<vector<float>>("max_size", R"code(Limit of the output size.
+
+When resizing uses the ``resize_shorter``, but not the"not_smaller", mode or leaving some extents
+unspecified, some images with high aspect ratios might produce extremely large outputs.
```

Comment:
```suggestion
When the operator is configured to keep aspect ratio and only the smaller dimension is specified, the other(s) can grow very large. This can happen when using ``resize_shorter`` argument or "not_smaller" mode or even with "default" mode when some dimensions are left unspecified.
```

Reply:
Done